src/codegen/formatted_elz_new/mod.rs

```diff
@@ -0,0 +1,184 @@
+                    s.push_str(add_blank(",", " ", code_to_char[i + 1]).as_str());
+                }
+
+                //add blank in the pre and next position
+                "+" => {
+                    s.push_str(add_blank("+", code_to_char[i - 1], code_to_char[i + 1]).as_str());
+                }
+                "-" => {
+                    s.push_str(add_blank("-", code_to_char[i - 1], code_to_char[i + 1]).as_str());
+                }
+                "*" => {
+                    s.push_str(add_blank("*", code_to_char[i - 1], code_to_char[i + 1]).as_str());
+                }
+                "/" => {
+                    if code_to_char[i + 1] == "/" {
+                        s.push_str("//");
+                        comment_line = true;
+                        i += 1;
+                    } else {
+                        s.push_str(add_blank("/", code_to_char[i - 1], code_to_char[i + 1]).as_str());
+                    }
+                }
+                "=" => {
+                    s.push_str(add_blank("=", code_to_char[i - 1], code_to_char[i + 1]).as_str());
+                    past_symbol = true;
+                }
+                ">" => {
+                    s.push_str(add_blank(">", code_to_char[i - 1], code_to_char[i + 1]).as_str());
+                }
+                "<" => {
+                    if code_to_char[i + 1] == ":" {
+                        s.push_str(add_blank("<", code_to_char[i - 1], " ").as_str());
+                    } else {
+                        s.push_str(add_blank("<", code_to_char[i - 1], code_to_char[i + 1]).as_str());
+                    }
+                }
+                "\\" => {
+                    if code_to_char[i + 1] == "n" {
+                        i += 1;
+                    }
+                }
+
+                _ => {
+                    if semicolon_symbol {
+                        if code_to_char[i] != " " {
+                            s.push_str(code_to_char[i]);
+                            semicolon_symbol = false;
+                        }
+                    } else {
+                        if code_to_char[i] == " " && check_symbol_behind_space(code_to_char[i + 1]) {} 
+                        else {
+                            s.push_str(code_to_char[i]);
+                        }
+                    }
+                }
+            }
+        }
+        i += 1;
+    }
+    println!("{}", s);
+    s
+}
+
+fn add_tab(tab: i32) -> String {
+    let mut count = 0i32;
+    let mut s = String::from("");
+    loop {
+        count += 1;
+        if count > tab {
+            break;
+        }
+        s.push_str("  ");
+    }
+    s
+}
+
+fn add_blank(c: &str, pre: &str, next: &str) -> String {
+    let mut s = String::from("");
+    if pre != " " {
+        s.push_str(" ");
+    }
+    s.push_str(c);
+    if next != " " {
+        s.push_str(" ");
+    }
+    s
+}
+
+fn check_symbol_behind_space(symbol: &str) -> bool {
+    let mut b = false;
```

Comment:
Don't do this, return value directly.

For example

if symbol == a { true }
if...
if...
else { false }